substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVSymbolSectionImpl.java

```diff
@@ -0,0 +1,201 @@
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.objectfile.pecoff.cv;
+
+import org.graalvm.compiler.debug.DebugContext;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static com.oracle.objectfile.pecoff.cv.CVConstants.CV_SIGNATURE_C13;
+import static com.oracle.objectfile.pecoff.cv.CVConstants.CV_SYMBOL_SECTION_NAME;
+
+public final class CVSymbolSectionImpl extends CVSectionImpl {
+
+    private static final int CV_VECTOR_DEFAULT_SIZE = 200;
+    private static final int CV_STRINGTABLE_DEFAULT_SIZE = 200;
+
+    private CVDebugInfo cvDebugInfo;
+    private CVFileRecord fileRecord;
+
+    private ArrayList<CVSymbolRecord> cvRecords = new ArrayList<>(CV_VECTOR_DEFAULT_SIZE);
+    private CVStringTable stringTable = new CVStringTable(CV_STRINGTABLE_DEFAULT_SIZE);
+
+    CVSymbolSectionImpl(CVDebugInfo cvDebugInfo) {
+        this.cvDebugInfo = cvDebugInfo;
+    }
+
+    @Override
+    public String getSectionName() {
+        return CV_SYMBOL_SECTION_NAME;
+    }
+
+    /*
+     * the CodeView symbol section ("debug$S") is actually a list of records containing sub-records
+     */
+    @Override
+    public void createContent(DebugContext debugContext) {
+        int pos = 0;
+        // enableLog(debugContext, pos);
+        debug = false;
+        log(debugContext, "CVSymbolSectionImpl.createContent() adding records");
+        addRecords(debugContext);
+        log(debugContext, "CVSymbolSectionImpl.createContent() start");
+        /* add header size */
+        pos += Integer.BYTES;
+        /* add sum of all record sizes */
+        for (CVSymbolRecord record : cvRecords) {
+            pos = CVUtil.align4(pos);
+            verboseLog(debugContext, "  [0x%08x] %s", pos, record.toString());
```

Comment:
There is no point calling verboseLog when you have set debug = false directly above.

n.b. In the ELF create/writeContent methods that field stops shared utility methods logging data at both createContent and writeContent time. But your create/write methods just cut and pastes the same inline code. So, you should either omit the log call under createContent or put the common code into a utility method called from both.

Reply:
the variable 'debug' can be unset at development time; some decisions are made then, so I need a way to see what's happening while working on this code. Logging in createContent() is uninteresting to the end user, but very interesting to the developer.  I've cleaned it up a bit, and I'm open to suggestions that allow me to turn logging during the createContent() phase on and off, but I'm going to need to log this phase until the type information code has been added.

Reply:
Well, logging stuff of interest only to the developer is part of what the debug context code is meant to do. That's why it supports a hierarchical naming scheme for identifying the context. If I want to enable debug during content creation I pass

    -H:Log=DebugCodeInfo:3 -Dgraal.LogFile=$PWD/out.log

Level 3 enables verbose logging. Keyword DebugCodeInfo matches every log context created under the call to installDebugInfo. If I only wanted log output for processing a specific method I could append the specific method name to the installDebugInfo keyword (with '.' separator).

Reply:
I've simplified the log enablement code to be entirely controlled by the debugContext, and if finer-grained control over loggin ends up being required, will ensure it only uses that mechanism, not compile-time contants.